test/e2e_federation/federated-namespace.go

```diff
@@ -97,18 +98,67 @@ var _ = framework.KubeDescribe("Federation namespace [Feature:Federation]", func
 			By(fmt.Sprintf("Verified that namespaces were not deleted from underlying clusters"))
 		})
 
+		// See https://github.com/kubernetes/kubernetes/issues/38225
+		It("deletes replicasets in the namespace when the namespace is deleted", func() {
+			fedframework.SkipUnlessFederated(f.ClientSet)
+
+			nsName := createNamespace(f.FederationClientset.Core().Namespaces())
+			rsName := v1.SimpleNameGenerator.GenerateName(replicaSetNamePrefix)
+			replicaCount := int32(2)
+			rs := &v1beta1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      rsName,
+					Namespace: nsName,
+				},
+				Spec: v1beta1.ReplicaSetSpec{
+					Replicas: &replicaCount,
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"name": "myrs"},
+					},
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{"name": "myrs"},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name:  "nginx",
+									Image: "nginx",
+								},
+							},
+						},
+					},
+				},
+			}
+
+			By(fmt.Sprintf("Creating replicaset %s in namespace %s", rsName, nsName))
+			_, err := f.FederationClientset.Extensions().ReplicaSets(nsName).Create(rs)
+			if err != nil {
+				framework.Failf("Failed to create replicaset %v in namespace %s, err: %s", rs, nsName, err)
+			}
+
+			By(fmt.Sprintf("Deleting namespace %s", nsName))
+			deleteNamespace(nil, nsName,
+				f.FederationClientset.Core().Namespaces().Get,
+				f.FederationClientset.Core().Namespaces().Delete)
+
+			By(fmt.Sprintf("Verify that replicaset %s was deleted as well", rsName))
+
+			waitForReplicaSetToBeDeletedOrFail(f.FederationClientset, nsName, rsName)
```

Comment:
@nikhiljindal makes sense.

@csbell good idea.

@perotinus is this something that you can fix?